pkg/resource/plugin/provider_plugin.go

```diff
@@ -404,49 +404,60 @@ func (p *provider) Read(urn resource.URN, id resource.ID, props resource.Propert
 	// Get the RPC client and ensure it's configured.
 	client, err := p.getClient()
 	if err != nil {
-		return nil, err
+		return nil, resource.StatusUnknown, err
 	}
 
 	// If the provider is not fully configured, return an empty bag.
 	if !p.cfgknown {
-		return resource.PropertyMap{}, nil
+		return resource.PropertyMap{}, resource.StatusUnknown, nil
 	}
 
 	// Marshal the input state so we can perform the RPC.
 	marshaled, err := MarshalProperties(props, MarshalOptions{Label: label, ElideAssetContents: true})
 	if err != nil {
-		return nil, err
+		return nil, resource.StatusUnknown, err
 	}
 
 	// Now issue the read request over RPC, blocking until it finished.
+	var readID resource.ID
+	var liveObject *_struct.Struct
+	var resourceError error
+	var resourceStatus = resource.StatusOK
 	resp, err := client.Read(p.ctx.Request(), &pulumirpc.ReadRequest{
 		Id:         string(id),
 		Urn:        string(urn),
 		Properties: marshaled,
 	})
 	if err != nil {
+		resourceStatus, readID, liveObject, resourceError = parseError(err)
 		logging.V(7).Infof("%s failed: %v", label, err)
-		return nil, err
+
+		if resourceStatus == resource.StatusUnknown {
```

Comment:
This should be `if resourceStatus != resourceStatus.StatusPartialFailure`

Reply:
Let's change this for `Create` and `Update` too then.